packages/core/src/State.ts

```diff
@@ -291,6 +291,8 @@ export function getPersistedState<TState extends AnyState>(
   const { configuration, transitions, tags, machine, children, ...jsonValues } =
     state;
 
+  const contextJSON = JSON.stringify(state.context);
+
   const childrenJson: Partial<PersistedMachineState<any>['children']> = {};
 
   for (const id in children) {
```

Comment:
we should throw when an inline actor is being persisted since we won't be able to restore it later

Reply:
How do we detect these again?

Reply:
don't we keep `src` on the `ActorRef`? I think that it makes to only keep there strings that reference actors in `implementations`. If we'd do that then differentiating an inline and a referenced actor would be super simple